perf_test/blas/blas1/KokkosBlas_team_dot_perf_test.cpp

```diff
@@ -0,0 +1,232 @@
+                "Vector is not a "
+                "Kokkos::View.");
+
+  using Scalar          = typename Vector::non_const_value_type;
+  using execution_space = typename Vector::execution_space;
+  typedef typename Kokkos::TeamPolicy<execution_space> team_policy;
+  typedef typename team_policy::member_type team_member;
+
+  // Declare Kokkos::View Vectors, x and y
+  Vector x;
+  Vector y;
+
+  KOKKOS_INLINE_FUNCTION void operator()(const team_member& team) const {
+    KokkosBlas::Experimental::dot(team, x, y);
+  }
+  // Constructor
+  teamDotFunctor<Vector>(Vector X_, Vector Y_) {
+    x = X_;
+    y = Y_;
+  }
+};
+
+template <class ExecSpace>
+void run(int m, int repeat) {
+  // Declare type aliases
+  using Scalar   = double;
+  using MemSpace = typename ExecSpace::memory_space;
+
+  // For the Team implementation of dot; ExecSpace is implicit;
+  using policy      = Kokkos::TeamPolicy<ExecSpace>;
+
+  // Create 1D view w/ Device as the ExecSpace; this is an input vector
+  Kokkos::View<Scalar*, MemSpace> x("X", m);
+  // Create 1D view w/ Device as the ExecSpace; this is the output vector
+  Kokkos::View<Scalar*, MemSpace> y("Y", m);
+
+  // Here, deep_copy is filling / copying values into Host memory from Views X
+  // and Y
+  Kokkos::deep_copy(x, 3.0);
+  Kokkos::deep_copy(y, 2.0);
+
+  std::cout << "Running BLAS Level 1 Kokkos Teams-based implementation DOT "
+               "performance experiment ("
+            << ExecSpace::name() << ")\n";
+
+  std::cout << "Each test input vector has a length of " << m << std::endl;
+
+  // Warm up run of dot:
+  teamDotFunctor<Kokkos::View<Scalar*, MemSpace>> teamDotFunctorWarmUpInstance(
+      x, y);
+
+  Kokkos::parallel_for("TeamDotDemoUsage -- Warm Up Run",
+                       policy(1, Kokkos::AUTO), teamDotFunctorWarmUpInstance);
+
+  Kokkos::fence();
+  Kokkos::Timer timer;
+
+  // Live test of dot:
+
+  teamDotFunctor<Kokkos::View<Scalar*, MemSpace>>
+      teamDotFunctorLiveTestInstance(x, y);
+  Kokkos::parallel_for("TeamDotDemoUsage -- Live Test", policy(1, Kokkos::AUTO),
+                       teamDotFunctorLiveTestInstance);
+
+  ExecSpace().fence();
+
+  // Kokkos Timer set up and data capture
+  double total = timer.seconds();
+  double avg   = total / repeat;
+  // Flops calculation for a 1D matrix dot product per test run;
+  size_t flopsPerRun = (size_t)2 * m;
+  printf("Avg DOT time: %f s.\n", avg);
+  printf("Avg DOT FLOP/s: %.3e\n", flopsPerRun / avg);
+}
+
+int main(int argc, char** argv) {
+  Params params;
+
+  if (parse_inputs(params, argc, argv)) {
+    return 1;
+  }
+
+  const int device_id = params.use_cuda - 1;
+
+  const int num_threads = std::max(params.use_openmp, params.use_threads);
+
+  Kokkos::initialize(Kokkos::InitArguments(num_threads, -1, device_id));
+
+  bool useThreads = params.use_threads != 0;
+  bool useOMP     = params.use_openmp != 0;
+  bool useCUDA    = params.use_cuda != 0;
+
+  bool useSerial = !useOMP && !useCUDA;
```

Comment:
Should add ``!useThreads`` to this condition

Reply:
@brian-kelley -- I corrected useSerial.